python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -550,6 +550,7 @@ def run_request_for_partition(
         partition_key: str,
         run_key: Optional[str],
         tags: Optional[Mapping[str, str]] = None,
+        job_name: Optional[str] = None,        
```

Comment:
I think we don't want to make this passable like so - I think we would just want to pass the current job's name to the run request.